packages/next/shared/lib/router/utils/prepare-destination.ts

```diff
@@ -49,17 +49,26 @@ export function matchHas(
     if (!hasItem.value && value) {
       params[getSafeParamName(key!)] = value
       return true
-    } else if (value) {
+    } else if (typeof value !== 'undefined') {
       const matcher = new RegExp(`^${hasItem.value}$`)
-      const matches = Array.isArray(value)
-        ? value.slice(-1)[0].match(matcher)
-        : value.match(matcher)
+      // a value of false is equivalent to ''
+      // e.g. /path?q - q is false and should be matched as ''
+      const match = (v?: string) => String(v || '').match(matcher)
+
+      let matches: ReturnType<typeof match> = null
+
+      if (Array.isArray(value)) {
+        // any match in an array means we have it
+        value.some((v) => (matches = match(v)))
```

Comment:
Should we match the last item here instead of the first?

Reply:
I'm not exactly sure what would be expected in this case

If you have multiple proxies which each add a header:
![Screenshot from 2022-02-09 16-52-58](https://user-images.githubusercontent.com/11590024/153296414-ee4a0120-d631-40a9-8182-1b8ff03bea1f.png)

If you have `[{ type: 'header', key: 'header', value: 'match-.+' }]` do you expect it to return `match-valueA` or `match-valueB` (`dontmatch-valueC` is ignored because it doesn't match)

Reply:
I went ahead and changed to select the last in an array, it seems to make more sense to me